### PR TITLE
New: Elementor admin - beta tester newsletter opt in

### PR DESCRIPTION
**Pull request by nuritsha**

### CODE REVIEW

File: assets/dev/js/admin/admin.js [hunk 1]
```diff
@@ -201,6 +203,33 @@
...
+
+				elementorCommon.dialogsManager.createWidget( 'lightbox', {
+					id: 'elementor-beta-tester-modal',
+					headerMessage: self.elements.$betaTesterDialogHeader,
+					message: self.elements.$betaTesterDialogForm,
+					closeButton: true,
+					closeButtonClass: 'eicon-close',
+
+					onReady: function() {
+						DialogsManager.getWidgetType( 'lightbox' ).prototype.onReady.apply( this, arguments );
```
Comment (matipojo) [inline assets/dev/js/admin/admin.js]: Why override the default?

File: assets/dev/js/admin/admin.js [hunk 2]
```diff
@@ -201,6 +203,33 @@
...
+					},
+				} ).show();
+			} );
+
+			self.elements.$betaTesterDialogForm.on( 'submit', function() {
+				const $email = $( '#elementor-beta-tester-email' ).val();
```
Comment (matipojo) [inline assets/dev/js/admin/admin.js]: Remove the `$` it's not an element.

File: core/admin/admin.php [hunk 3]
```diff
@@ -551,6 +551,57 @@ public function elementor_dashboard_overview_widget() {
...
+	 *
+	 * Display a dialog box to suggest the user to opt-in to the beta testers newsletter.
+	 *
+	 * Fired by `admin_footer` filter.
+	 *
+	 * @since 2.6.0
+	 * @access public
+	 */
+	public function print_beta_tester_dialog() {
+
```
Comment (matipojo) [inline core/admin/admin.php]: Remove line

File: includes/user.php [hunk 4]
```diff
@@ -232,6 +247,27 @@ public static function set_introduction_viewed( array $data ) {
 		update_user_meta( get_current_user_id(), self::INTRODUCTION_KEY, $user_introduction_meta );
 	}
 
+	public static function register_as_beta_tester( array $data ) {
+
+		add_filter( 'https_ssl_verify', '__return_false' );
```
Comment (matipojo) [inline includes/user.php]: Remove.

File: core/admin/admin.php [hunk 5]
```diff
@@ -551,6 +551,57 @@ public function elementor_dashboard_overview_widget() {
...
+						echo __( 'Want to be the first to hear about new features & software improvements? leave your email below', 'elementor' );
+						?>
+					</div>
+					<div class="elementor-beta-tester-dialog-input-wrapper">
+						<input id="elementor-beta-tester-email" class="elementor-beta-tester-dialog-input" name="beta_tester_email" type="email" required value="<?php echo $current_user->user_email; ?>" />
+						<input type="submit" class="elementor-button elementor-button-success" value="<?php echo esc_attr__( 'Sign Up', 'elementor' ); ?>">
+					</div>
+					<div class="beta-tester-terms">
+						<?php
+						echo sprintf( '%s<a href="https://elementor.com/terms/">%s</a> %s <a href="https://elementor.com/privacy-policy/">%s</a>', __( 'By entering your email, you agree to Elementor\'s ', 'elementor' ), __( 'Terms of Service', 'elementor' ), __( 'and', 'elementor' ), __( 'Privacy Policy', 'elementor' ) );
```
Comment (shilo-ey) [inline core/admin/admin.php]: change terms url to > https://go.elementor.com/beta-testers-newsletter-terms/
change privacy policy url to > https://go.elementor.com/beta-testers-newsletter-privacy/

File: includes/user.php [hunk 6]
```diff
@@ -24,6 +24,20 @@ class User {
...
+	 * API URL.
+	 *
+	 * Holds the URL of the Beta Tester Opt-in API.
+	 *
+	 * @since 1.0.0
+	 * @access private
+	 *
+	 * @var string API URL.
+	 */
+	private static $_api_url = 'https://my.elementor.com/api/v1/beta_tester/';
```
Comment (matipojo) [inline includes/user.php]: Use a const.

File: assets/dev/js/admin/admin.js [hunk 7]
```diff
@@ -201,6 +203,33 @@
...
+				elementorCommon.ajax.addRequest( 'beta_tester_confirmed', {
```
Comment (matipojo) [inline assets/dev/js/admin/admin.js]: Rename `beta_tester_newsletter`.

**Review by matipojo — CHANGES_REQUESTED** [review]

File: assets/dev/js/admin/admin.js (hunk 1 above)
Comment (nuritsha) [reply assets/dev/js/admin/admin.js]: removed.

File: assets/dev/js/admin/admin.js (hunk 2 above)
Comment (nuritsha) [reply assets/dev/js/admin/admin.js]: removed.

File: core/admin/admin.php (hunk 3 above)
Comment (nuritsha) [reply core/admin/admin.php]: removed.

File: includes/user.php (hunk 6 above)
Comment (nuritsha) [reply includes/user.php]: done.

File: assets/dev/js/admin/admin.js (hunk 7 above)
Comment (nuritsha) [reply assets/dev/js/admin/admin.js]: done.

File: includes/user.php (hunk 4 above)
Comment (nuritsha) [reply includes/user.php]: for sure? this exists in all other elementor-api calls

File: core/admin/admin.php [hunk 8]
```diff
@@ -593,7 +592,7 @@ public function print_beta_tester_dialog() {
 					</div>
 					<div class="beta-tester-terms">
 						<?php
-						echo sprintf( '%s<a href="https://elementor.com/terms/">%s</a> %s <a href="https://elementor.com/privacy-policy/">%s</a>', __( 'By entering your email, you agree to Elementor\'s ', 'elementor' ), __( 'Terms of Service', 'elementor' ), __( 'and', 'elementor' ), __( 'Privacy Policy', 'elementor' ) );
+						echo sprintf( '%s<a href="https://go.elementor.com/beta-testers-newsletter-terms">%s</a> %s <a href="https://go.elementor.com/beta-testers-newsletter-privacy">%s</a>', __( 'By entering your email, you agree to Elementor\'s ', 'elementor' ), __( 'Terms of Service', 'elementor' ), __( 'and', 'elementor' ), __( 'Privacy Policy', 'elementor' ) );
```
Comment (matipojo) [inline core/admin/admin.php]: const

**Review by matipojo — CHANGES_REQUESTED** [review]

**Review by matipojo — APPROVED** [review]

**Pull request merged**